client/utils/guidance.py

```diff
@@ -0,0 +1,210 @@
+from client.protocols.common import models
+from client.utils import auth
+from urllib.request import urlopen
+from client.utils import assess_id_util
+import json
+import os
+import sys
+
+"""
+This utility is called by unlock.py. This guidance utility changes the message that students see
+after they input in wrong answers. If a student guesses a certain amount of wrong answers
+that all show a certain type of confusion about the problem, this utility will instead of showing
+the default "Not Quite Try Again" message will show some kind of message that will target 
+that type of misunderstanding. 
+
+This utility object requires internet access to determine what treatmeng group they are assigned
+to. The different treatment groups will have varying threshold level of answers as well as different
+messages and other differences. It will contact the server defined below in the variable TGSERVER with
+the user's email and lab assignment to get the treatment group number. 
+
+Commonly used acronyms: 
+TG = treatment group number
+KI = Type of targeted understanding
+misU = Type of misunderstanding the student is showing
+"""
+
+TGSERVER = "http://tg-server.app.cs61a.org/"
+LOCAL_TG = "tests/tg.ok_tg"
+OK_GUIDANCE = "tests/.ok_guidance"
+DEFAULT_MSG = "-- Not quite. Try again! --"
+DEFAULT_RETURN_VALUE = ({}, -1, "")
```

Comment:
The default message seems to indicate that the student answered a question incorrectly, so a possible name for this would be `GUIDANCE_INFO_WHEN_INCORRECT`. My concern is that there are several methods in this class, so `RETURN_VALUE` is ambiguous. It'd be better if the name described what the triplet `(dict, int, str)` represents (e.g. "guidance info" or "guidance result", but perhaps you can think of a more accurate term).